cms/navigation/tests/test_navigation_tags.py

```diff
@@ -0,0 +1,82 @@
+import wagtail.coreutils
+from django.test import TestCase
+
+from cms.navigation.templatetags.navigation_tags import main_menu_columns, main_menu_highlights
+from cms.navigation.tests.factories import (
+    ColumnBlockFactory,
+    HighlightsBlockFactory,
+    MainMenuFactory,
+    SectionBlockFactory,
+    TopicLinkBlockFactory,
+)
+
+
+class MainMenuTemplateTagTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.mock_request = wagtail.coreutils.get_dummy_request()
+        cls.main_menu = MainMenuFactory()
+
+        highlights = [{"type": "highlight", "value": HighlightsBlockFactory()}] * 3
+
+        topic_links = [TopicLinkBlockFactory()] * 5
+
+        sections = [SectionBlockFactory(links=topic_links)] * 3
+
+        column_value = ColumnBlockFactory(sections=sections)
+
+        columns = [
+            {
+                "type": "column",
+                "value": column_value,
+            }
+        ] * 3
+
+        cls.main_menu.highlights = highlights
+        cls.main_menu.columns = columns
```

Comment:
Issue: These values won't be reflected in the database versions of these rows - you might need to call `cls.main_menu.save`.

Reply:
I don't think we need to call `cls.main_menu.save` in this case for these particular tests. As I am just passing the in-memory `cls.main_menu` object to `main_menu_highlights` and `main_menu_columns`. These functions operate on the object I give them, so the updated `highlights` and `columns` are available without ever touching the database.

But to be honest, because of the reasons I mentioned above, I think it's a good practice/ good to do `.save()` reflecting the state we would expect if this were a real usage with queries hitting the database.

So thanks for this as I'll go with this suggestion so that in the future, any code that re-fetches objects from the database won't break etc.

Reply:
Done in def5c07.
Resolving. Let's get this in